### PR TITLE
[FEAT] Terminal Integration

### PR DESCRIPTION
**Pull request by hv7214**
**Note**
There is one issue, typing `exit` in the terminal closes the terminal widget which in turn results in blank Terminal. Soon I will apply the patch for the same : ) 

**Issue**
#13

### CODE REVIEW

**Comment by hv7214** [comment]
@JaDogg

File: PyRun.pro [hunk 1]
```diff
@@ -5,7 +5,7 @@
 #   - Bhathiya Perera
 #-------------------------------------------------
 
-QT       += core gui
+QT       += core gui network
```
Comment (JaDogg) [inline PyRun.pro]: Do we need the network?

Comment (hv7214) [reply PyRun.pro]: yeah forgot to remove it!

**Review by JaDogg — CHANGES_REQUESTED** [review]
This cannot be compiled on Windows as you are including header in various places. Only import if it's not windows. :)

File: UI/mainview.h [hunk 2]
```diff
@@ -10,6 +10,8 @@
 #include <QInputDialog>
 #include <QThread>
 #include <QApplication>
+#include <qtermwidget5/qtermwidget.h>
```
Comment (JaDogg) [inline UI/mainview.h:14]: Only include this if it's not windows

File: UI/mainview.h [hunk 3]
```diff
@@ -108,10 +109,13 @@ class MainView : public QMainWindow {
     Snippets *m_snippets;
     XTute *m_tute;
     QCompleter *completer;
+    QTermWidget* terminal;
```
Comment (JaDogg) [inline UI/mainview.h:115]: Don't create this object if it's windows

File: UI/mainview.cpp [hunk 4]
```diff
@@ -129,6 +132,24 @@ void MainView::SetupHighlighter() {
...
 
+void MainView::SetupTerminal() {
+#ifndef Q_OS_WIN
+    setenv("TERM", "xterm-256color", 1);
+    SetTerminal();
+#endif
+}
+
+void MainView::SetTerminal() {
+    terminal = new QTermWidget();
```
Comment (JaDogg) [inline UI/mainview.cpp:144]: Don't do anything in this for windows.

**Comment by JaDogg** [comment]
Try to do these changes and it will compile on windows without terminal support :) 
```diff
diff --git a/UI/mainview.cpp b/UI/mainview.cpp
index fae56a2..6b1857c 100644
--- a/UI/mainview.cpp
+++ b/UI/mainview.cpp
@@ -140,14 +140,22 @@ void MainView::SetupTerminal() {
 }
 
 void MainView::SetTerminal() {
+#ifndef Q_OS_WIN
     terminal = new QTermWidget();
+#endif
+
 #ifdef Q_OS_MACX
     terminal->setKeyBindings("macbook");
-#else
+#endif
+
+#ifdef Q_OS_LINUX
     terminal->setKeyBindings("linux");
 #endif
+
+#ifndef Q_OS_WIN
     terminal->setColorScheme("Tango");
     ui->dwTerminal->setWidget(terminal);
+#endif
 }
 
 void MainView::SetCompleter(CodeEditor *editor) {
@@ -221,7 +229,10 @@ MainView::~MainView() {
     this->SaveContent();
     m_workerThread->quit();
     m_workerThread->wait();
+
+#ifndef Q_OS_WIN
     delete terminal;
+#endif
     delete m_workerThread;
     delete ui;
     delete m_tute;
diff --git a/UI/mainview.h b/UI/mainview.h
index 0585aa9..a2f78a9 100644
--- a/UI/mainview.h
+++ b/UI/mainview.h
@@ -10,7 +10,9 @@
 #include <QInputDialog>
 #include <QThread>
 #include <QApplication>
+#ifndef Q_OS_WIN
 #include <qtermwidget5/qtermwidget.h>
+#endif
 
 // internal
 #include "CodeEditor/pythonsyntaxhighlighter.h"
@@ -109,7 +111,9 @@ class MainView : public QMainWindow {
     Snippets *m_snippets;
     XTute *m_tute;
     QCompleter *completer;
+#ifndef Q_OS_WIN
     QTermWidget* terminal;
+#endif
     bool m_markTute = false;
     int m_markIndex = -1;
     void ChangeFontSize(QFont font, int size);

```

**Review by JaDogg — APPROVED** [review]

**Pull request merged**